### PR TITLE
ACS-3651 Validate node pameters.

### PR DESCRIPTION
**Pull request by tpage-alfresco**

### CODE REVIEW

**Comment by tpage-alfresco** [comment]
This checks the node exists and the user has read access.

File: remote-api/src/main/java/org/alfresco/rest/api/impl/rules/ActionParameterConverter.java [hunk 1]
```diff
@@ -56,13 +62,19 @@
     private final DictionaryService dictionaryService;
     private final ActionService actionService;
     private final NamespaceService namespaceService;
+    private final NodeService nodeService;
```
Comment (mpichura) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/rules/ActionParameterConverter.java:65]: I don't think this one is ever used.

File: remote-api/src/main/resources/alfresco/public-rest-context.xml [hunk 2]
```diff
@@ -910,6 +910,9 @@
         <constructor-arg name="actionService" ref="ActionService"/>
         <constructor-arg name="dictionaryService" ref="DictionaryService"/>
         <constructor-arg name="namespaceService" ref="NamespaceService"/>
+        <constructor-arg name="nodeService" ref="NodeService"/>
```
Comment (mpichura) [inline remote-api/src/main/resources/alfresco/public-rest-context.xml:913]: This one may be obsolete (nod used in the Java class).

**Review by mpichura — APPROVED** [review]
LGTM, some small cleanup needed.
Though I am surprised that E2E tests don't fail since this [PR ](https://github.com/Alfresco/alfresco-community-repo/pull/1496) is not yet merged which means that folders in rule actions are set as dummy node ids.  
Subject to green  build. Same (completely unrelated) annoying E2E tests failed in travis again :(

File: remote-api/src/main/resources/alfresco/public-rest-context.xml (hunk 2 above)
Comment (tpage-alfresco) [reply remote-api/src/main/resources/alfresco/public-rest-context.xml]: Thanks - good spot.

**Pull request merged**